src/gs-category-page.c

```diff
@@ -342,15 +342,28 @@ gs_category_page_create_filter_list (GsCategoryPage *self,
 			 * placeholder app tiles get cleared out, then set
 			 * "empty state" message for an empty USB disk
 			 */
-			if (g_strcmp0 (gs_category_get_id (category), "usb") == 0) {
-				gs_category_page_populate_filtered (self, s);
+			if (g_strcmp0 (gs_category_get_id (category), "usb") != 0) {
```

Comment:
You should be able to factor out these common calls to `gtk_widget_set_visible()` and `continue` with something like:

```
gboolean have_usb_apps = FALSE;
if (g_strcmp0 (gs_category_get_id (category), "usb") == 0) {
    				GPtrArray *usb_apps;
￼ 				category_is_usb = TRUE;
￼ 				gs_category_page_populate_filtered (self, s);
￼ 				/* After reloading the page it will get the actual apps for the
￼                                  * for the usb so we test if the category will truly have the 
￼                                  * no apps box.
￼                                  */
￼ 				usb_apps = gs_category_get_children (category);
￼ 				if (usb_apps != NULL && usb_apps->len >= 0)
￼ 				￼ 	  have_usb_apps = TRUE;
}

if (!have_usb_apps) {
					gtk_widget_set_visible (self->no_apps_box, category_is_usb);
					gtk_widget_set_visible (self->scrolledwindow_category, 
							!category_is_usb);
					continue;
}
```

(but fix the horrific indentation in my example)